### PR TITLE
fix transactions event emitter

### PR DESCRIPTION
**Pull request by talhamalik883**
transaction events emitter fix

### CODE REVIEW

File: packages/smart-account/src/SmartAccount.ts [hunk 1]
```diff
@@ -587,7 +587,7 @@ class SmartAccount extends EventEmitter {
     if (updateImplTrx.data != '0x') {
       transactions.unshift(updateImplTrx)
     }
-    const response = await aaSigner.sendTransactionBatch(transactions, false)
+    const response = await aaSigner.sendTransactionBatch(transactions, this)
```
Comment (livingrockrises) [inline packages/smart-account/src/SmartAccount.ts:590]: why false argument is changed to 'this'
wouldn't that mess up with the order?

**Review by livingrockrises — COMMENTED** [review]
reviewed. how sendTransaction and sendTransactionBatch is defined in 4337Signer class?

File: packages/smart-account/src/SmartAccount.ts (hunk 1 above)
Comment (talhamalik883) [reply packages/smart-account/src/SmartAccount.ts:590]: It was wrong initially, sendTransactionBatch function just take two arguments. One is transaction object and other is event emitter, initially false value was supplied that was wrong so have corrected that

Comment (livingrockrises) [reply packages/smart-account/src/SmartAccount.ts:590]: and sendTransaction takes three args now? without breaking ethers signer interface

Comment (talhamalik883) [reply packages/smart-account/src/SmartAccount.ts:590]: yes, here are the args

```
 transaction: TransactionRequest,
 walletDeployOnly = false,
 engine?: any // EventEmitter
```

**Comment by livingrockrises** [comment]
ok, can you please change the base to development. It won't directly be merged into master..

**Review by livingrockrises — APPROVED** [review]
LGTM

**Review by AmanRaj1608 — APPROVED** [review]

**Pull request merged**